dbcon/mysql/ha_mcs_execplan.cpp

```diff
@@ -8594,15 +8494,45 @@ int getGroupPlan(gp_walk_info& gwi, SELECT_LEX& select_lex, SCSEP& csep, cal_gro
             break;
 
         ptp = new ParseTree(new LogicOperator("and"));
-        //ptp->left(filters);
-        ptp->right(filters);
-        lhs = gwi.ptWorkStack.top();
+        ptp->left(filters);
```

Comment:
Why do you decide to swap the branches here?

Reply:
Table ordering is reverse in SELECT_LEX::join_list (which is what we use in buildJoin() now) compared to SELECT_LEX::table_list (which we used earlier). So I needed to swap the branches here to keep the filter ordering same as before.